test/suites/utilization/fips_test.go

```diff
@@ -0,0 +1,239 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utilization_test
+
+import (
+	"fmt"
+	"strings"
+
+	opstatus "github.com/awslabs/operatorpkg/status"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/karpenter/pkg/test"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1beta1"
+	"github.com/Azure/karpenter-provider-azure/pkg/utils"
+	"github.com/Azure/karpenter-provider-azure/pkg/utils/nodeclaim"
+)
+
+var _ = Describe("FIPS", func() {
+	Context("FIPS Validation", func() {
```

Comment:
Can we not test all of these validation tests with envtest?

Doesn't feel like something we need a real cluster for? 2/3 of these are rejected by apiserver (so definitely can be done in envtest). The 3rd feels like it should be pretty easily doable in envtest as well?

Thinking: E2E is great but we should keep it focused on the stuff we actually need E2E to cover. Not sold we do for these 3 tests.